packages/gatsby-source-wordpress/src/steps/ingest-remote-schema/build-queries-from-introspection/recursively-transform-fields.js

```diff
@@ -313,32 +313,15 @@ export function transformField({
         gatsbyNodesInfo.typeNames.includes(possibleType.name)
       )
 
-  const isAMediaItemNode = isAGatsbyNode && typeName === `MediaItem`
-
-  // pull the id and __typename for connections to media item gatsby nodes
-  if (isAMediaItemNode && hasIdField) {
+  if (isAGatsbyNode && hasIdField) {
     return {
       fieldName: fieldName,
       fields: [`__typename`, `id`],
       fieldType,
     }
-  } else if (isAGatsbyNode && hasIdField) {
-    const isAnInterfaceType =
-      // if this is an interface
-      typeKind === `INTERFACE` || fieldType.kind === `INTERFACE`
-
-    return {
-      fieldName: fieldName,
-      fields: isAnInterfaceType
-        ? // we need the typename for interfaces
-          [`id`, `__typename`]
-        : // or just the id for 1:1 connections to gatsby nodes
-          [`id`],
-      fieldType,
-    }
```

Comment:
All of this was just conditionally adding __typename to generated query fields, but now we just add it always for any kind of connection